storage/database.go

```diff
@@ -292,8 +308,13 @@ func (d *db) FetchBlocks(
 	starts []time.Time,
 ) ([]block.FetchBlockResult, error) {
 	callStart := d.nowFn()
-	n, err := d.namespaceFor(namespace)
+	if d.isOverloaded() {
```

Comment:
I went back and forth on this, the nice thing about putting it in the database layer is that the RPC layer doesn't need to know what status the database is in since the logic is encapsulated in the database layer, and load shedding can be disabled via setting the options if so desired. I do however see the argument for the other approach too.